### PR TITLE
SymbolsExtractor simplifications

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java [hunk 1]
```diff
@@ -263,7 +264,7 @@ public Expression visitJoin(JoinNode node, Void context)
         // Conjuncts without any symbol dependencies cannot be applied to the effective predicate (e.g. FALSE literal)
         return conjuncts.stream()
                 .map(expression -> pullExpressionThroughSymbols(expression, outputSymbols))
-                .map(expression -> SymbolsExtractor.extractAll(expression).isEmpty() ? TRUE_LITERAL : expression)
+                .map(expression -> SymbolsExtractor.extractAll(expression).anyMatch(alwaysTrue()) ? TRUE_LITERAL : expression)
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java]: This is wrong. `isEmpty()` can be migrated to `count() == 1`

File: presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolsExtractor.java [hunk 2]
```diff
@@ -77,9 +77,10 @@ private SymbolsExtractor() {}
 
     public static Stream<Symbol> extractAll(Expression expression)
     {
-        ImmutableList.Builder<Symbol> builder = ImmutableList.builder();
-        new SymbolBuilderVisitor().process(expression, builder);
-        return builder.build().stream();
+        return AstUtils.preOrder(expression)
+                .filter(SymbolReference.class::isInstance)
+                .map(SymbolReference.class::cast)
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolsExtractor.java]: you do not need to map

**Comment by findepi** [comment]
@kokosing , comments applied

**Comment by findepi** [comment]
@kokosing , i squashed commits locally and bisected travis failure to "Remove SymbolsExtractor.SymbolsExtractor" commit. I don't understand. I am aware that `SymbolsExtractor.SymbolsExtractor` extended `DefaultExpressionTraversalVisitor` which doesn't visit certain things, but it looks like this should NOT matter. Apparently it does.

**Pull request closed without merging**